client/client.go

```diff
@@ -1288,28 +1288,34 @@ type TSFuture interface {
 }
 
 func (req *tsoRequest) Wait() (physical int64, logical int64, err error) {
-	// If tso command duration is observed very high, the reason could be it
-	// takes too long for Wait() be called.
 	start := time.Now()
+	// `cmdDurationTSOAsyncWait` is useful to see how long a TSFuture takes
+	// until the Wait() function is called.
+	// It represents the duration between the TSO request and actually being used.
 	cmdDurationTSOAsyncWait.Observe(start.Sub(req.start).Seconds())
 	select {
 	case err = <-req.done:
 		err = errors.WithStack(err)
 		defer tsoReqPool.Put(req)
 		if err != nil {
+			// `cmdFailDurationTSO` is useful to see how long a TSO request takes until an error is met.
 			cmdFailDurationTSO.Observe(time.Since(req.start).Seconds())
-			return 0, 0, err
+		} else {
+			physical, logical = req.physical, req.logical
 		}
-		physical, logical = req.physical, req.logical
-		now := time.Now()
-		cmdDurationWait.Observe(now.Sub(start).Seconds())
-		cmdDurationTSO.Observe(now.Sub(req.start).Seconds())
-		return
 	case <-req.requestCtx.Done():
-		return 0, 0, errors.WithStack(req.requestCtx.Err())
+		err = errors.WithStack(req.requestCtx.Err())
 	case <-req.clientCtx.Done():
-		return 0, 0, errors.WithStack(req.clientCtx.Err())
-	}
+		err = errors.WithStack(req.clientCtx.Err())
+	}
+	now := time.Now()
+	// `cmdDurationWait` is useful to see how long it takes to Wait() a TSO result.
+	// It partially or fully represents the client-side pressure.
+	cmdDurationWait.Observe(now.Sub(start).Seconds())
```

Comment:
How do we distinguish between success and failed requests in the metrics?

Reply:
I think the problem here is that the old cmdDurationWait only shows the part of the durations, i.e, the successful part. This PR extends these metrics to all TSO durations. IMO, distinguishing success or not is not that important even unnecessary for this one. It just shows the whole picture as a TSO Wait indicator.